openhands/llm/llm.py

```diff
@@ -259,8 +262,11 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
                 )
                 kwargs['messages'] = messages
 
-                # add stop words if the model supports it
-                if self.config.model not in MODELS_WITHOUT_STOP_WORDS:
+                # add stop words if the model supports it and reasoning is not enabled
+                if (
+                    self.config.model not in MODELS_WITHOUT_STOP_WORDS
+                    and not self.enable_reasoning
+                ):
```

Comment:
Actually i'm confused -- we by default using reasoning effort high for all models that actually supports reasoning -- why do we need this tag though

Reply:
Hmm how about we maintain a centralized JSON that includes all these common model name mapping?

Like you could probably put these model name mapping in the consolidated JSON, instead of extra code?:
https://github.com/All-Hands-AI/OpenHands/issues/10262

Reply:
Tell you what... for now let's make it an environment variable. What do you want it called?